Dev

b-artem opened the pull request:
https://bookstore-artem.herokuapp.com/
Valid coupon code: 12345678

b-artem committed:
fix csrf issue and providing email

b-artem committed:
remove byebug

b-artem committed:
remove pry from code

b-artem committed:
add ruby version for heroku

b-artem committed:
syntax error

b-artem committed:
syntax error

b-artem committed:
minor changes

b-artem committed:
Merge pull request #3 from b-artem/fb_auth

Fb auth

b-artem committed:
add book

b-artem committed:
add book author category models

b-artem committed:
add first_ and last_name validations

b-artem committed:
Merge pull request #4 from b-artem/i18n

I18n - add uk locale to app layout and login page

b-artem committed:
add uk locale to app layout and login page

b-artem committed:
add login specs

b-artem committed:
add signup test

b-artem committed:
add login page view specs

b-artem committed:
add some controller specs for login page

b-artem committed:
add registartion specs

b-artem committed:
change database.yml

b-artem committed:
just test

b-artem committed:
few changes to devise controllers specs

b-artem committed:
Merge pull request #6 from b-artem/tests

Tests (few tests for devise)

b-artem committed:
home view

b-artem committed:
start editing book view

b-artem committed:
add decorator book#authors_names

b-artem committed:
feew changes to book#show view

b-artem committed:
start for read more link

b-artem committed:
home page bestsellers

b-artem committed:
latest on home page

b-artem committed:
start catalog view

b-artem committed:
catalog categories and sort_by functionality

b-artem committed:
read_more works now

b-artem committed:
start adding reviews

b-artem committed:
add stars to reviews

b-artem committed:
add rails admin for review state

b-artem committed:
Merge pull request #7 from b-artem/books2

Books2

b-artem committed:
add Cart and LineItem models

b-artem committed:
couple of specs for book#destroy

b-artem committed:
add line_item#create

b-artem committed:
add to cart btn and input on book#show

b-artem committed:
partials for line_items visible or hiddden xs

b-artem committed:
continue cart update

b-artem committed:
try cart ajax

b-artem committed:
add cart destroy

b-artem committed:
Merge pull request #8 from b-artem/cart

Cart

b-artem committed:
few minor changes to view

b-artem committed:
add AJAX to cart

b-artem committed:
add models for checkout

b-artem committed:
fix models

b-artem committed:
add link to cart icon in app layout

b-artem committed:
minor change to view

b-artem committed:
start checkout form

b-artem committed:
few minor changes

b-artem committed:
add orders controller

b-artem committed:
add simple_form

b-artem committed:
continue form object

b-artem committed:
work on wizard

b-artem committed:
change to simple_form

b-artem committed:
need to fix one single form

b-artem committed:
continue form object

b-artem committed:
changes to db

b-artem committed:
add use_billing_address_as_shipping column to Orders

b-artem committed:
address form implementation

b-artem committed:
two forms

b-artem committed:
order form_object works now

b-artem committed:
addresses form improvements

b-artem committed:
use billing address logic

b-artem committed:
continue checkout address page

b-artem committed:
minor change to js checkout address page

b-artem committed:
checkout/address page works

b-artem committed:
checkout delivery page

b-artem committed:
checkout/payment page

b-artem committed:
checkout/confirm page

b-artem committed:
order params now set up

b-artem committed:
checkout/delivery loads shipping method

b-artem committed:
checkout/complete page

b-artem committed:
add payment setup

b-artem committed:
circle.yml add

b-artem committed:
comment figaro requires keys in omniauth.rb for circle

b-artem committed:
change login & signup specs a bit

b-artem committed:
comment unnecessary specs

b-artem committed:
add few Book specs

b-artem committed:
few more Book specs

b-artem committed:
start Author specs

b-artem committed:
few Author specs

b-artem committed:
add specs fo LineItem and Cart

b-artem committed:
start review scopes specs

b-artem committed:
add Review specs

b-artem committed:
add Order and Category specs

b-artem committed:
add few model specs

b-artem committed:
Merge pull request #9 from b-artem/checkout

Checkout

b-artem committed:
FormObject specs

b-artem committed:
fix PaymentForm factory

b-artem committed:
just test

b-artem committed:
start Home page specs

b-artem committed:
add drivers for js

b-artem committed:
testing change driver to selenium

b-artem committed:
set default js driver to webkit

b-artem committed:
temp fix for controllers #new specs

b-artem committed:
add wait_for_ajax module

b-artem committed:
fix some specs

b-artem committed:
start best sellers specs

b-artem committed:
add scope best_seller for Book

b-artem committed:
best_seller now returns book

b-artem committed:
continue best sellers specs

b-artem committed:
fix spec

b-artem committed:
continue best sellers specs

b-artem committed:
fix some specs

b-artem committed:
fix specs

b-artem committed:
finish Home page specs

b-artem committed:
start Cart specs

b-artem committed:
start cart specs

b-artem committed:
finish edit cart specs

b-artem committed:
finish Cart specs

b-artem committed:
start add discount

b-artem committed:
add coupon logic, change checkout

b-artem committed:
db rollback

b-artem committed:
before migrate

b-artem committed:
after migrate

b-artem committed:
fix specs after changing some Order methods

b-artem committed:
settings

b-artem committed:
Merge pull request #10 from b-artem/reattach

Reattach

b-artem committed:
continue settings

b-artem committed:
finish orders

b-artem committed:
remove shoulda temporarily for CI

b-artem committed:
return shoulda for CI

b-artem committed:
pending settings controller

b-artem committed:
create simple seed

b-artem committed:
seed

b-artem committed:
seed2

b-artem committed:
db restore

b-artem committed:
remove pry for heroku

b-artem committed:
order authorization

b-artem committed:
temp fix controller spec

b-artem committed:
Merge pull request #11 from b-artem/settings

Settings

b-artem committed:
start aws

b-artem committed:
make images

b-artem committed:
Merge pull request #12 from b-artem/images

make images

b-artem committed:
delete commemts

b-artem committed:
books views

b-artem committed:
book delete odd

b-artem committed:
cart delete odd actions

b-artem committed:
home cart devise delete odd

b-artem committed:
delete odd line_items views

b-artem committed:
delete line item odd actions

b-artem committed:
delete odd specs

b-artem committed:
fix delete specs

b-artem committed:
delete odd from orders

b-artem committed:
fix orders specs

b-artem committed:
Merge pull request #13 from b-artem/specs2

Specs2

b-artem committed:
few changes to checkout views

b-artem committed:
checkout views few changes

b-artem committed:
Merge pull request #15 from b-artem/specs3

Specs3

b-artem committed:
javascript now fires once

b-artem committed:
delete user works now

b-artem committed:
checkout rescue from incorrect path

b-artem committed:
start populat scope

b-artem committed:
fix address_form factory

b-artem committed:
fix fb login

b-artem committed:
Merge pull request #14 from b-artem/specs3

Specs3

b-artem committed:
minor changes in layout

b-artem committed:
change alert color

b-artem committed:
pagination change

b-artem committed:
book filters

b-artem committed:
first

b-artem committed:
s3 url

b-artem committed:
start test with s3

b-artem committed:
add stub for images[] in feature specs

b-artem committed:
add stub for images[] for feature specs 2

b-artem committed:
minor changes

b-artem committed:
fix specs

b-artem committed:
minor changes

b-artem committed:
Merge pull request #17 from b-artem/finish_s3

Finish s3

b-artem committed:
cart icon counter change

b-artem committed:
double action trouble

b-artem committed:
fix turbolinks fired actions twice

b-artem committed:
fix bug turbolinks fired actions twice

b-artem committed:
few changes

b-artem committed:
precompile stars

b-artem committed:
precompile assets in production

b-artem committed:
clean assets.rb a bit

b-artem committed:
Merge pull request #18 from b-artem/few_changes

Few changes, fix bug - turbolinks fired up twice

Dulenchik commented on app/assets/javascripts/books.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
@b-artem Let's deleted blank js file

b-artem replied:
Should I also delete all auto-generated blank folders? E.g. assets/javascripts/channels

Dulenchik commented on app/assets/javascripts/carts.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
@b-artem Let's deleted blank js file

Dulenchik commented on app/assets/javascripts/checkouts.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
@b-artem Let's deleted blank js file

Dulenchik commented on app/assets/javascripts/home.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
@b-artem Let's deleted blank js file

Dulenchik commented on app/assets/javascripts/line_items.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
@b-artem Let's deleted blank js file

Dulenchik commented on app/assets/javascripts/reviews.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
@b-artem Let's deleted blank js file

Dulenchik commented on app/assets/javascripts/settings.coffee:
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
@b-artem Let's deleted blank js file

Dulenchik commented on app/assets/stylesheets/books.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Books controller here.
```
@b-artem Let's deleted blank css file

Dulenchik commented on app/assets/stylesheets/carts.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Carts controller here.
```
@b-artem Let's deleted blank css file

Dulenchik commented on app/assets/stylesheets/checkouts.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the checkouts controller here.
```
@b-artem Let's deleted blank css file

Dulenchik commented on app/assets/stylesheets/home.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Home controller here.
```
@b-artem Let's deleted blank css file

Dulenchik commented on app/assets/stylesheets/line_items.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the LineItems controller here.
```
@b-artem Let's deleted blank css file

Dulenchik commented on app/assets/stylesheets/review.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Review controller here.
```
@b-artem Let's deleted blank css file

Dulenchik commented on app/assets/stylesheets/settings.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Settings controller here.
```
@b-artem Let's deleted blank css file

Dulenchik commented on app/controllers/application_controller.rb:
```diff
@@ -1,3 +1,56 @@
+  end
+
+  private
+    # Its important that the location is NOT stored if:
```
@b-artem Let's remove these comments from codebase

b-artem replied:
May I keep auto-generated comments? E.g. in config/initializers/devise.rb:
```ruby
  # ==> Mailer Configuration
  # Configure the e-mail address which will be shown in Devise::Mailer,
  # note that it will be overwritten if you use your own mailer class...

Dulenchik commented on app/assets/javascripts/books/filter.js:
```diff
@@ -0,0 +1,12 @@
+$(document).on('turbolinks:load', function() {
+  $("#sort-by li a").click(function() {
+    console.log('filter');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/assets/javascripts/books/read_more.js:
```diff
@@ -0,0 +1,4 @@
+readMore = function(full_description) {
+  $(event.target).parent().html(full_description);
+  console.log('read_more');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/assets/javascripts/line_items/decrement_quantity.js:
```diff
@@ -0,0 +1,13 @@
+  }
+  input.val(quantity);
+  input.trigger("change");
+  console.log('decrement_quantity');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/assets/javascripts/line_items/increment_quantity.js:
```diff
@@ -0,0 +1,8 @@
+  quantity++;
+  input.val(quantity);
+  input.trigger("change");
+  console.log('increment_quantity');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/assets/javascripts/line_items/update_quantity.js:
```diff
@@ -0,0 +1,4 @@
+updateQuantity = function(lineItemId) {
+  $("#update-line-item-" + lineItemId).click();
+  console.log('update_quantity');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/assets/javascripts/orders/checkouts/use_billing_address.js:
```diff
@@ -0,0 +1,11 @@
+    var checked = $(this).prop('checked')
+    $('#shipping-address-form').prop('hidden', checked);
+  });
+  console.log('use_billing_address');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/assets/javascripts/orders/orders/filter.js:
```diff
@@ -0,0 +1,7 @@
+  $("#state-filter li a").click(function() {
+    var text = ($(this).text());
+    setTimeout(function() { $("span#text").text(text); }, 300);
+    console.log('filter');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/assets/javascripts/reviews/raty.js:
```diff
@@ -0,0 +1,7 @@
+    path: '/assets',
+    scoreName: 'review[score]'
+  });
+  console.log('raty');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/assets/javascripts/settings/remove_account.js:
```diff
@@ -0,0 +1,7 @@
+    $("#remove-account-btn").prop('disabled', !$("#remove-account-btn").prop('disabled'));
+    $("#remove-account-btn").toggleClass('disabled');
+  });
+  console.log('remove_account');
```
@b-artem Let's remove logs from client-side js

Dulenchik commented on app/controllers/application_controller.rb:
```diff
@@ -1,3 +1,56 @@
+    if I18n.available_locales.map(&:to_s).include?(params[:locale])
+      I18n.locale = params[:locale]
+    else
+      flash.now[:notice] = "#{params[:locale]} translation not available"
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/carts_controller.rb:
```diff
@@ -0,0 +1,35 @@
+        format.html { redirect_to @cart, notice: 'Coupon was successfully applied' }
+        format.json { render :show, status: :ok, location: @cart }
+      else
+        format.html { redirect_to @cart, alert: "There is no such coupon or it's not valid anymore" }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/carts_controller.rb:
```diff
@@ -0,0 +1,35 @@
+    respond_to do |format|
+      if coupon
+        @cart.update(coupon: coupon)
+        format.html { redirect_to @cart, notice: 'Coupon was successfully applied' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/concerns/current_cart.rb:
```diff
@@ -0,0 +1,21 @@
+    end
+
+    def ensure_cart_isnt_empty
+      redirect_to(books_url, notice: 'Your cart is empty') unless @cart.line_items.any?
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/concerns/current_order.rb:
```diff
@@ -0,0 +1,20 @@
+        Order.find(session[:order_id])
+      end
+    rescue ActiveRecord::RecordNotFound
+      flash[:alert] = "Order #{session[:order_id]} was not found"
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,58 @@
+    respond_to do |format|
+      if @line_item.save
+        format.html { redirect_back fallback_location: root_path,
+                      notice: 'Line item was successfully added to a cart' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,58 @@
+                      notice: 'Line item was successfully added to a cart' }
+        format.json { render :show, status: :created, location: @line_item }
+      else
+        format.html { redirect_back fallback_location: root_path, alert: 'Line item was not created' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,58 @@
+    end
+  rescue
+    redirect_back fallback_location: root_path,
+                  alert: 'Line item was not added. Please enter positive integer quantity'
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,58 @@
+  def update
+    respond_to do |format|
+      if @line_item.update(line_item_params)
+        format.html { redirect_to @line_item, notice: 'Line item was successfully updated.' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,58 @@
+  def destroy
+    @line_item.destroy
+    respond_to do |format|
+      format.html { redirect_to @cart, notice: 'Product was successfully removed' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/orders/orders_controller.rb:
```diff
@@ -0,0 +1,68 @@
+  def update
+    respond_to do |format|
+      if @order.update(order_params)
+        format.html { redirect_to @order, notice: 'Order was successfully updated.' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/orders/orders_controller.rb:
```diff
@@ -0,0 +1,68 @@
+  def destroy
+    @order.destroy
+    respond_to do |format|
+      format.html { redirect_to orders_url, notice: 'Order was successfully destroyed.' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/reviews_controller.rb:
```diff
@@ -0,0 +1,61 @@
+    respond_to do |format|
+      if @review.save
+        format.html { redirect_to @review.book,
+                      notice: 'Thanks for Review. It will be published as soon as Admin will approve it.' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/reviews_controller.rb:
```diff
@@ -0,0 +1,61 @@
+  def update
+    respond_to do |format|
+      if @review.update(review_params)
+        format.html { redirect_to @review, notice: 'Review was successfully updated.' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/reviews_controller.rb:
```diff
@@ -0,0 +1,61 @@
+  def destroy
+    @review.destroy
+    respond_to do |format|
+      format.html { redirect_to root_path, notice: 'Review was successfully destroyed.' }
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/settings_controller.rb:
```diff
@@ -0,0 +1,81 @@
+  def update_email
+    if @user.update(user_params)
+      redirect_to settings_edit_path(tab: 'privacy'),
+                  notice: 'Your email was successfully changed'
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/settings_controller.rb:
```diff
@@ -0,0 +1,81 @@
+                  notice: 'Your email was successfully changed'
+    else
+      redirect_to settings_edit_path(tab: 'privacy'),
+                  alert: "Your email wasn't changed"
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/settings_controller.rb:
```diff
@@ -0,0 +1,81 @@
+    if @user.update_with_password(user_params)
+      bypass_sign_in(@user)
+      redirect_to settings_edit_path(tab: 'privacy'),
+                  notice: "Your password was successfully changed"
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/settings_controller.rb:
```diff
@@ -0,0 +1,81 @@
+                  notice: "Your password was successfully changed"
+    else
+      redirect_to settings_edit_path(tab: 'privacy'),
+                  alert: "Your password wasn't changed"
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/settings_controller.rb:
```diff
@@ -0,0 +1,81 @@
+      redirect_to settings_edit_path, notice: 'Address was successfully updated'
+    else
+      redirect_to settings_edit_path(settings_params),
+                  alert: "Address wasn't updated"
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/users/omniauth_callbacks_controller.rb:
```diff
@@ -0,0 +1,14 @@
+  def facebook
+    @user = User.from_omniauth(request.env["omniauth.auth"])
+    unless request.env["omniauth.auth"].info.email
+      flash[:warning] = "No email provided by facebook. Custom email created. Please change it in settings"
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/users/omniauth_callbacks_controller.rb:
```diff
@@ -0,0 +1,14 @@
+  end
+
+  def failure
+    flash[:warning] = 'Failed to sign in'
```
@b-artem Let's add this flash message to I18n

Dulenchik commented on app/controllers/books_controller.rb:
```diff
@@ -0,0 +1,43 @@
+class BooksController < ApplicationController
+  before_action :set_book, only: [:show]
```
@b-artem it should be processed via cancancan using `load_and_authorize_resource`

b-artem replied:
Done

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,58 @@
+class LineItemsController < ApplicationController
+  include CurrentCart
+  before_action :set_cart, only: [:create, :destroy]
+  before_action :set_line_item, only: [:update, :destroy]
```
@b-artem it should be processed via cancancan using `load_and_authorize_resource`

Dulenchik commented on app/controllers/orders/orders_controller.rb:
```diff
@@ -0,0 +1,68 @@
+class Orders::OrdersController < ApplicationController
+  include CurrentCart
```
@b-artem you already have this module included in `ApplicationController`, you don't need to include it in class which was inherited from `ApplicationController`

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,58 @@
+class LineItemsController < ApplicationController
+  include CurrentCart
```
@b-artem you already have this module included in `ApplicationController`, you don't need to include it in class which was inherited from `ApplicationController`

Dulenchik commented on app/controllers/carts_controller.rb:
```diff
@@ -0,0 +1,35 @@
+class CartsController < ApplicationController
+  include CurrentCart
```
@b-artem you already have this module included in `ApplicationController`, you don't need to include it in class which was inherited from `ApplicationController`

Dulenchik commented on app/controllers/carts_controller.rb:
```diff
@@ -0,0 +1,35 @@
+class CartsController < ApplicationController
+  include CurrentCart
+  # before_action :set_cart, only: [:show, :update]
```
@b-artem let's remove this commented line

Dulenchik commented on app/decorators/address_decorator.rb:
```diff
@@ -0,0 +1,15 @@
+    city + ' ' + zip
+  end
+
+  def phone
```
@b-artem this method is redundant because it doesn't provide any additional logic, you just delegate it to `object`. You can remove it, `delegate_all` does the same

Dulenchik commented on app/decorators/checkout_decorator.rb:
```diff
@@ -0,0 +1,3 @@
+class CheckoutDecorator < Draper::Decorator
```
@b-artem Let's remove this blank decorator

Dulenchik commented on app/decorators/setting_decorator.rb:
```diff
@@ -0,0 +1,3 @@
+class SettingDecorator < Draper::Decorator
```
@b-artem Let's remove this blank decorator

Dulenchik commented on app/helpers/books_helper.rb:
```diff
@@ -0,0 +1,5 @@
+module BooksHelper
+  def latest(quantity = 3)
+    Book.order('created_at DESC').limit(quantity)
```
@b-artem it's vey bad idea to use activerecord in helpers. If you need data in template - load it in controller

b-artem replied:
Done

Dulenchik commented on app/helpers/carts_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module CartsHelper
```
@b-artem Let's remove this blank helper module

Dulenchik commented on app/helpers/checkouts_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module CheckoutsHelper
```
@b-artem Let's remove this blank helper module

Dulenchik commented on app/helpers/home_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module HomeHelper
```
@b-artem Let's remove this blank helper module

Dulenchik commented on app/helpers/line_items_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module LineItemsHelper
```
@b-artem Let's remove this blank helper module

Dulenchik commented on app/helpers/orders_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module OrdersHelper
```
@b-artem Let's remove this blank helper module

Dulenchik commented on app/helpers/reviews_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module ReviewsHelper
```
@b-artem Let's remove this blank helper module

Dulenchik commented on app/helpers/settings_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module SettingsHelper
```
@b-artem Let's remove this blank helper module

Dulenchik commented on app/helpers/shipping_methods_helper.rb:
```diff
@@ -0,0 +1,5 @@
+module ShippingMethodsHelper
+  def shipping_methods
+    ShippingMethod.order('price ASC')
```
@b-artem it's vey bad idea to use activerecord in helpers. If you need data in template - load it in controller

b-artem replied:
Done

b-artem replied:
Done

Dulenchik commented on app/models/address.rb:
```diff
@@ -0,0 +1,7 @@
+  belongs_to :order, optional: true
+
+  validates :type, inclusion: { in: %w(BillingAddress ShippingAddress),
+    message: "%{value} is not a valid type" }
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/author.rb:
```diff
@@ -0,0 +1,8 @@
+
+  validates :first_name, :last_name,
+            presence: true,
+            format: { with: /\A[a-zA-Z]+\z/, message: 'Only allows letters' },
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/book.rb:
```diff
@@ -0,0 +1,65 @@
+  validates :title, uniqueness: { case_sensitive: false }
+  validates :image_url, allow_blank: true, format: {
+    with: %r{\.(gif|jpg|png)\Z}i,
+    message: 'must be a URL for GIF, JPG or PNG image.'
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/coupon.rb:
```diff
@@ -0,0 +1,13 @@
+  validates :code, :discount, :valid_until, presence: true
+  validates :code,
+            format: { with: /\A[a-zA-Z0-9]+\z/,
+                      message: 'Only allows letters and numbers' },
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/forms/address_form.rb:
```diff
@@ -0,0 +1,50 @@
+            :phone, presence: true, if: :need_shipping_address?
+  validates :type, inclusion: { in: %w(BillingAddress ShippingAddress) }
+  validates :first_name, :last_name,
+            format: { with: /\A[a-zA-Z]+\z/, message: 'Only allows letters' },
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/forms/address_form.rb:
```diff
@@ -0,0 +1,50 @@
+            length: { maximum: 49 }, if: :need_shipping_address?
+  validates :address,
+            format: { with: /\A[A-Za-z0-9, -]+\z/,
+                      message: "Only allows a-z, A-Z, 0-9,’,’, ‘-’, ‘ ’" },
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/forms/address_form.rb:
```diff
@@ -0,0 +1,50 @@
+            length: { maximum: 49 }, if: :need_shipping_address?
+  validates :city, :country,
+            format: { with: /\A[A-Z a-z]+\z/,
+                      message: "Only allows a-z, ' ', A-Z" },
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/forms/address_form.rb:
```diff
@@ -0,0 +1,50 @@
+            length: { maximum: 49 }, if: :need_shipping_address?
+  validates :zip,
+            format: { with: /\A[0-9-]+\z/,
+                      message: "Only allows 0-9 '-'" },
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/forms/address_form.rb:
```diff
@@ -0,0 +1,50 @@
+            length: { maximum: 10 }, if: :need_shipping_address?
+  validates :phone,
+            format: { with: /\A\+{1}[0-9]+\z/,
+                      message: "Only allows '+', 0-9" },
```
@b-artem Let's make error message to be translated using I18n

Dulenchik commented on app/models/forms/address_form.rb:
```diff
@@ -0,0 +1,50 @@
+    true
+  end
+
+  # def save
```
@b-artem let's remove this commented code

Dulenchik commented on app/models/forms/order_form.rb:
```diff
@@ -0,0 +1,30 @@
+  validates :billing_address, presence: true
+  validates :shipping_address, presence: true, if: :need_shipping_address?
+
+  # def initialize(*args)
```
@b-artem let's remove this commented code

Dulenchik commented on app/views/devise/registrations/new.html.haml:
```diff
@@ -0,0 +1,34 @@
+.text-center
+  %h3.general-login-title= t '.sign_up'
+  -# %a.general-login-icon{href: "#"}
```
@b-artem let's remove this commented code

Dulenchik commented on app/views/devise/registrations/new.html.haml:
```diff
@@ -0,0 +1,34 @@
+
+  .form-group.mb-30
+    = f.label :password, t('devise.password'), class: "control-label input-label"
+    -# - if @minimum_password_length
```
@b-artem let's remove this commented code

Dulenchik commented on app/views/kaminari/_gap.html.haml:
```diff
@@ -0,0 +1,8 @@
+-#  Non-link tag that stands for skipped pages...
```
@b-artem let's remove these commented lines

Dulenchik commented on app/views/kaminari/_paginator.html.haml:
```diff
@@ -0,0 +1,18 @@
+-#  The container tag
```
@b-artem Let's remove these commented lines

Dulenchik commented on app/views/kaminari/_prev_page.html.haml:
```diff
@@ -0,0 +1,9 @@
+-#  Link to the "Previous" page
```
@b-artem let's remove this blank file

Dulenchik commented on app/views/kaminari/_page.html.haml:
```diff
@@ -0,0 +1,10 @@
+-#  Link showing page number
```
@b-artem let's remove this blank file

Dulenchik commented on app/views/kaminari/_next_page.html.haml:
```diff
@@ -0,0 +1,9 @@
+-#  Link to the "Next" page
```
@b-artem let's remove this blank file

Dulenchik commented on app/views/kaminari/_last_page.html.haml:
```diff
@@ -0,0 +1,9 @@
+-#  Link to the "Last" page
```
@b-artem let's remove this blank file

Dulenchik commented on app/views/kaminari/_first_page.html.haml:
```diff
@@ -0,0 +1,9 @@
+-#  Link to the "First" page
```
@b-artem let's remove this blank file

Dulenchik commented on app/views/orders/checkouts/_steps.html.haml:
```diff
@@ -0,0 +1,35 @@
+        %li.step-divider
+
+-#
+-# %h1.general-title-margin Checkout
```
@b-artem Let's remove these commented lines

Dulenchik commented on app/views/reviews/_review.html.haml:
```diff
@@ -0,0 +1,56 @@
+          %i.fa.fa-star.rate-star
+      %p= review.text
+
+  -# .reviews
```
@b-artem Let's remove these commented lines

Dulenchik commented on app/views/reviews/create.html.haml:
```diff
@@ -0,0 +1,2 @@
+%h1 Review#create
```
@b-artem let's remove this dummy template

Dulenchik commented on app/views/reviews/edit.html.haml:
```diff
@@ -0,0 +1,2 @@
+%h1 Review#edit
```
@b-artem let's remove this dummy template

Dulenchik commented on app/views/reviews/update.html.haml:
```diff
@@ -0,0 +1,2 @@
+%h1 Review#update
```
@b-artem let's remove this dummy template

Dulenchik commented on config/application.rb:
```diff
@@ -27,5 +28,11 @@ class Application < Rails::Application
+
+    config.generators do |g|
+      g.test_framework :rspec
+      # g.fixture_replacement :factory_girl, dir: 'spec/factories'
```
@b-artem let's remove this commented line

Dulenchik commented on circle.yml:
```diff
@@ -0,0 +1,100 @@
+    version:
+      2.4.1
+
+# dependencies:
```
@b-artem do we need this commented example? if no - let's remove it

Dulenchik reviewed (changes requested):
@b-artem To summarize:

- let's remove all comments, logs, and blank files;
- let's start to use I18n everywhere (controllers, views, error messages, etc);

I'll continue to review the actual implementation when code will be cleaned up

b-artem committed:
assets

b-artem committed:
controllers comments, i18n

b-artem committed:
decorators, reviews controller

b-artem committed:
decorators, helpers, models

b-artem committed:
views except devise

b-artem committed:
some configs

b-artem committed:
specs

b-artem committed:
remove old image_url from books

b-artem committed:
devise

b-artem committed:
i18n in specs

b-artem committed:
Merge pull request #19 from b-artem/clean_up

Clean up

b-artem committed:
remove double include CurrentCart

b-artem commented:
@Dulenchik I've cleaned up code. Haven't change implementation yet. I have two questions:
1. Should I also delete all auto-generated blank folders? E.g. assets/javascripts/channels
2. May I keep auto-generated comments? E.g. in config/initializers/devise.rb:
```ruby
  # ==> Mailer Configuration
  # Configure the e-mail address which will be shown in Devise::Mailer,
  # note that it will be overwritten if you use your own mailer class...

b-artem committed:
add cancancan step1

b-artem committed:
cancancan to Order

Dulenchik commented on app/controllers/books_controller.rb:
```diff
@@ -0,0 +1,43 @@
+      @book = Book.find(params[:id])
+    end
+
+    def book_params
```
@b-artem this method isn't used, you can remove it

b-artem replied:
Done

Dulenchik commented on app/controllers/carts_controller.rb:
```diff
@@ -0,0 +1,33 @@
+
+  private
+
+    def cart_params
```
@b-artem this method isn't used as well

b-artem replied:
Done

Dulenchik commented on app/controllers/carts_controller.rb:
```diff
@@ -0,0 +1,33 @@
+  end
+
+  def update
+    coupon = Coupon.find_by_code(params[:cart][:coupon][:code])
```
@b-artem note that `find_by_*` methods are deprecated, rails suggests to use `find_by` method. 

Is it real case when coupon param key wasn't pass? and we have `undefined method '[]' for nil class`?

b-artem replied:
@Dulenchik Use find_by method now.
When user doesn't fill in coupon code field, `params[:cart][:coupon][:code] == ""`, and `coupon == nil`. I check if it's true and flash corresponding message

Dulenchik commented on app/controllers/concerns/current_cart.rb:
```diff
@@ -0,0 +1,21 @@
+    end
+
+    def ensure_cart_isnt_empty
+      redirect_to(books_url, notice: t('cart_is_empty')) unless @cart.line_items.any?
```
@b-artem it will be better to replace `any?` with `exists?`. Please check this article: https://www.ombulabs.com/blog/benchmark/performance/rails/present-vs-any-vs-exists.html

b-artem replied:
Thanks, done

Dulenchik commented on app/controllers/concerns/current_cart.rb:
```diff
@@ -0,0 +1,21 @@
+
+    def destroy_cart
+      Cart.destroy(session[:cart_id])
+      session[:cart_id] = nil
```
@b-artem if you want to delete something from session - use delete instead of nil assignment, because assignment doesn't remove key from session hash

https://stackoverflow.com/questions/3995816/what-is-the-best-way-to-clear-a-session-variable-in-rails

b-artem replied:
Done

Dulenchik commented on app/controllers/concerns/current_order.rb:
```diff
@@ -0,0 +1,20 @@
+    end
+
+    def empty_current_order
+      session[:order_id] = nil
```
@b-artem the same comment about clean up session storage as above

b-artem replied:
Done

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,57 @@
+  def create
+    book = Book.find(params[:book_id])
+    @line_item = @cart.add_product(book, params[:quantity])
+    @line_item.price = book.price
```
@b-artem since you build line_item using `add_product` method and pass book as a product, you can make this price assignment in `add_product` method

b-artem replied:
Done

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,57 @@
+      @line_item = LineItem.find(params[:id])
+    end
+
+    def line_item_params
```
@b-artem this method looks unused

b-artem replied:
@Dulenchik, `set_line_item` isn't used now because `load_and_authorize_resource`.
`line_item_params` is used in update action, although one of parameters wasn't used, deleted it.

Dulenchik commented on app/controllers/line_items_controller.rb:
```diff
@@ -0,0 +1,57 @@
+class LineItemsController < ApplicationController
+  before_action :set_cart, only: [:create, :destroy]
+  before_action :set_line_item, only: [:update, :destroy]
```
@b-artem Don't forget about cancancan (load_and_authorize_resource)

b-artem replied:
Done

Dulenchik commented on app/controllers/orders/checkouts_controller.rb:
```diff
@@ -0,0 +1,101 @@
+    end
+
+    def clear_payment_data
+      session[:card_number] = nil
```
@b-artem the same comment about clean-up session

b-artem replied:
Done

Dulenchik commented on app/controllers/orders/checkouts_controller.rb:
```diff
@@ -0,0 +1,101 @@
+
+  def show
+    @current_order = current_order
+    case step
```
@b-artem this method looks extremely complex and should be refactored

Dulenchik commented on app/controllers/orders/checkouts_controller.rb:
```diff
@@ -0,0 +1,101 @@
+    render_wizard
+  end
+
+  def update
```
@b-artem the same comment as above: too complex method and needs refactoring/reorganization

Dulenchik commented on app/controllers/orders/orders_controller.rb:
```diff
@@ -0,0 +1,67 @@
+  authorize_resource
+
+  def index
+    @orders = current_user.orders.send(params[:state]).order('completed_at DESC')
```
@b-artem usage of `send`, based on value from params is extremely dangerous, if you need flexibility, you should:

- forget about the `send` and use `public_send`;
- whitelist available values for state param;

UPDATED: I scrolled down and checked `sanitize_filter_param`. It means that you already filtered state value - :+1: . My two suggestions will be:

- use `public_send`;
- make the filtering more obvious, for example, stop to sanitize param in `before_action`, instead define method which returns allowed state;

b-artem replied:
@Dulenchik, changed id. Method `state_filter` returns allowed state. Besides, filtering uses `where` now instead of `public_send`

Dulenchik commented on app/controllers/orders/orders_controller.rb:
```diff
@@ -0,0 +1,67 @@
+      @order = Order.find(params[:id])
+    end
+
+    def order_params
```
@b-artem never used method

b-artem replied:
Done

Dulenchik commented on app/controllers/orders/orders_controller.rb:
```diff
@@ -0,0 +1,67 @@
+  def show
+  end
+
+  def create
```
@b-artem this method is a good candidate for service object

b-artem replied:
Done

b-artem replied:
@Dulenchik, done it. Although, I'm not sure if it's correct.

Dulenchik commented on app/decorators/order_decorator.rb:
```diff
@@ -0,0 +1,29 @@
+  end
+
+  def completed_at
+    object.completed_at.strftime("%B %e, %Y")
```
@b-artem since people from different countries use different formats for dates, times etc, I18n helps us with localization of these values: http://guides.rubyonrails.org/i18n.html#adding-date-time-formats

So, let's stop to format date/time manually, instead create rule in I18n and start to use it

b-artem replied:
Done

Dulenchik commented on app/decorators/order_decorator.rb:
```diff
@@ -0,0 +1,29 @@
+
+  def completed_at_full_date
+    return I18n.t('orders.orders.index.not_completed') unless object.completed_at
+    object.completed_at.strftime("%Y-%m-%d")
```
@b-artem the same comment about localization

b-artem replied:
Done

Dulenchik commented on app/decorators/review_decorator.rb:
```diff
@@ -0,0 +1,7 @@
+  delegate_all
+
+  def date
+    created_at.strftime("%d/%m/%y")
```
@b-artem the same comment about localization

b-artem replied:
Done

Dulenchik commented on app/decorators/shipping_method_decorator.rb:
```diff
@@ -0,0 +1,11 @@
+  delegate_all
+
+  def terms
+    "#{days_min} to #{days_max} days"
```
@b-artem let's translate it

b-artem replied:
Done

Dulenchik commented on app/decorators/order_decorator.rb:
```diff
@@ -0,0 +1,29 @@
+  end
+
+  def state
+    return I18n.t('orders.orders.filter.waiting_for_processing') if object.state == 'in_queue'
```
@b-artem aasm provides you methods, according to current state of object, so you can use `object.in_queue?`

b-artem replied:
I got it. However, no need to check `object.in_queue?` now, as I simplified method. Besides, now it has all translations.

Dulenchik commented on app/models/book.rb:
```diff
@@ -0,0 +1,61 @@
+class Book < ApplicationRecord
+  CATEGORIES = %w[mobile_development photo web_design web_development].freeze
```
@b-artem it looks extremely strange, why do you need these hard-coded categories? you have categories as a model in app

b-artem replied:
No hard-coded categories now

Dulenchik commented on app/models/book.rb:
```diff
@@ -0,0 +1,61 @@
+  private
+
+    def ensure_not_referenced_by_any_line_item
+      unless line_items.empty?
```
@b-artem the same comment as for `any?`

b-artem replied:
Done

Dulenchik commented on app/models/book.rb:
```diff
@@ -0,0 +1,61 @@
+  end
+
+  scope :popular_first_ids, -> do
+    return Book.none unless LineItem.any?
```
@b-artem the same comment as about `any?` above

b-artem replied:
Done

Dulenchik commented on app/models/book.rb:
```diff
@@ -0,0 +1,61 @@
+
+  scope :best_seller, ->(category) do
+    return Book.none unless CATEGORIES.include?(category.to_s)
+    return Book.public_send(category).first unless LineItem.any?
```
@b-artem the same comment as about any? above

b-artem replied:
Done

Dulenchik commented on app/models/book.rb:
```diff
@@ -0,0 +1,61 @@
+
+  paginates_per 12
+
+  scope :mobile_development, -> do
```
@b-artem these scopes also look strange, what should we do if we need to add one more category about 'DB architecture' for example

b-artem replied:
@Dulenchik, deleted these scopes. `Category.find_by(name: category_name).books` does the same

Dulenchik commented on app/models/cart.rb:
```diff
@@ -0,0 +1,31 @@
+  end
+
+  def subtotal
+    return 0 unless line_items.any?
```
@b-artem the same comment about `any?` as before

b-artem replied:
Done

Dulenchik commented on app/models/cart.rb:
```diff
@@ -0,0 +1,31 @@
+
+  def subtotal
+    return 0 unless line_items.any?
+    line_items.sum(&:subtotal)
```
@b-artem Prefer to use sql query over enumerable methods

b-artem replied:
@Dulenchik, `subtotal` isn't present in DB, it's instance method only. Is it better to use SQL anyway or it should be presented in DB?

Dulenchik commented on app/models/cart.rb:
```diff
@@ -0,0 +1,31 @@
+  end
+
+  def products_quantity
+    return 0 unless line_items.any?
```
@b-artem the same comment about `any?`

b-artem replied:
Done

Dulenchik commented on app/models/cart.rb:
```diff
@@ -0,0 +1,31 @@
+
+  def products_quantity
+    return 0 unless line_items.any?
+    line_items.sum(&:quantity)
```
@b-artem the same comment about `sum`

Dulenchik commented on app/models/forms/payment_form.rb:
```diff
@@ -0,0 +1,26 @@
+            length: { minimum: 3 }
+
+  def save
+    return true if valid?
```
@b-artem maybe just return `valid?`

b-artem replied:
Done

Dulenchik commented on app/models/order.rb:
```diff
@@ -0,0 +1,51 @@
+  end
+
+  def subtotal
+    return 0 unless line_items.any?
```
@b-artem the same comment about `any?`

b-artem replied:
Done

Dulenchik commented on app/models/order.rb:
```diff
@@ -0,0 +1,51 @@
+
+  def subtotal
+    return 0 unless line_items.any?
+    line_items.sum(&:subtotal)
```
@b-artem the same comment about `sum`

Dulenchik commented on app/models/order.rb:
```diff
@@ -0,0 +1,51 @@
+
+  def discount
+    return 0 unless coupon
+    -1 * subtotal * coupon.discount / 100
```
@b-artem  remember our discussion about this `-1`. Let's really move it in decorator and make `discount` value positive by default

b-artem replied:
Done

Dulenchik commented on app/models/order.rb:
```diff
@@ -0,0 +1,51 @@
+  private
+
+    def generate_number
+      number = id.to_s.prepend('R' + '0' * (8 - id.to_s.length))
```
@b-artem interesting implementation, I'd recommend to check this method for string: https://ruby-doc.org/core-2.2.3/String.html#method-i-rjust

b-artem replied:
Thanks, done

Dulenchik commented on app/models/review.rb:
```diff
@@ -0,0 +1,34 @@
+    end
+  end
+
+  scope :new_, -> do
```
@b-artem first of all, really weird name for scope. Why do you need it? aasm provides helper methods, so you can use `Review.unprocessed`

UPDATED: I can't find usage of this scope, is it true?

b-artem replied:
@Dulenchik, this scope is used in rails_admin.rb:
```ruby
  config.model 'Review' do
    list do
      scopes [:new_, :processed]
      ...
```
The idea behind that name was to meet the following requirement:
* Admin sees the list of new reviews(unprocessed) in **New** tab.

It's easy to change model or field label in Rails Admin, but I still can't figure out how to change scope label from 'Unpocessed' to 'New'

Dulenchik commented on app/models/user.rb:
```diff
@@ -0,0 +1,26 @@
+      if auth.info.email
+        user.email = auth.info.email
+      else
+        user.email = user.info.first_name + user.info.first_name + '@fb.com'
```
@b-artem double first_name?)

b-artem replied:
It was a misspelling), correct now

Dulenchik reviewed (changes requested):
@b-artem New portion of feedback, please check usage of your methods and deleted dead-code. 
Also please, do something with categories, if you have any questions or need a discussion - just ping me.
Don't forget about changes for cancancan which we discussed.

Next time I'll check the views + js

b-artem committed:
some improvements according to review

b-artem committed:
few more changes

b-artem committed:
fix specs

b-artem committed:
few more changes

b-artem committed:
start rewrite categories

Dulenchik commented:
@b-artem about your questions - as you wish. 
Ideally if you don't use action cable - channels should be removed from app but now it isn't super important.
About the comments in initializers, as I said, it's totally up to you

b-artem committed:
rewrite #best_seller

b-artem committed:
fix specs

b-artem committed:
start service object for OrdersController#create

b-artem committed:
start specs for checkout

b-artem committed:
views changes

b-artem committed:
Merge pull request #20 from b-artem/improvements

Improvements

b-artem commented:
@Dulenchik, I've corrected most cases for now. Except refactoring/reorganization `checkout_controller`. It's still in progress along with specs.

b-artem committed:
minor change

b-artem committed:
checkout address specs

Dulenchik commented on app/assets/javascripts/books/read_more.js:
```diff
@@ -0,0 +1,3 @@
+readMore = function(full_description) {
```
@b-artem This function is defined in global scope. Let's isolate it like you did for other pages

b-artem replied:
Done

Dulenchik commented on app/assets/javascripts/line_items/update_quantity.js:
```diff
@@ -0,0 +1,3 @@
+updateQuantity = function(lineItemId) {
```
@b-artem This function is defined in global scope. Let's isolate it like you did for other pages

b-artem replied:
Done

Dulenchik reviewed (changes requested):
@b-artem Right now some functions in js exist in global scope. It's a bad practice. Also, please change your `onclick` handlers in html with jQuery `click` events.

Let me know when you finish refactoring of ruby parts and redeploy app on Heroku

b-artem committed:
save adresses specs

b-artem committed:
finish checkout saving addresses specs

b-artem committed:
start delivery specs

b-artem committed:
finish Checkout delivery specs

b-artem committed:
finish Checkout payment specs

b-artem committed:
start Checkout Confirm step specs

b-artem committed:
fix country selecting

b-artem committed:
finish confirm step specs

b-artem committed:
few minor changes

b-artem committed:
finish checkout specs

b-artem committed:
fix specs

b-artem committed:
refactor checkout_controller step 1

b-artem committed:
finish refactor checkout

b-artem committed:
start refactor js

b-artem committed:
finish refactor js

b-artem committed:
Merge pull request #21 from b-artem/refactoring

Changes according to code review (checkout controller refactoring + specs, js refactoring) and few more changes

b-artem commented:
@Dulenchik, I finished refactoring Checkout controller after adding specs for it. Made changes to js according to the last review and changed onclick handlers in html with jQuery click events.

App was redeployed on Heroku after the above changes

UPDATED: added specs for other controllers
UPDATED: replaced JS with CoffeeScript 
UPDATED: adjusted mailer, added devise confirmable to users

b-artem committed:
start controller specs

b-artem committed:
cart and home controllers specs

b-artem committed:
orders_controller specs

b-artem committed:
reviews_controller specs

b-artem committed:
line_items_controller specs

b-artem committed:
Merge pull request #22 from b-artem/controller_specs

Controller specs

b-artem committed:
replace it_behaves_like with include_examples

b-artem committed:
fix specs

b-artem committed:
coffee time

b-artem committed:
coffee time

b-artem committed:
delete unnecessary 'returns' from CoffeeScript

b-artem committed:
config production 1

b-artem committed:
change host settings

b-artem committed:
fix host settings

b-artem committed:
change domain in config

b-artem committed:
delete port from default_url_options

b-artem committed:
devise mailer adjustment, rewrite devise/passwords edit and new views

b-artem committed:
clean up mailer

b-artem committed:
little fix for address_form factory

b-artem committed:
add confirmable to users

b-artem committed:
change user factory to support confirmable

b-artem committed:
fix signup_spec to support confirmable

b-artem committed:
allow unconfirmed access for 2 days

b-artem committed:
Merge pull request #23 from b-artem/mailer

Adjust mailer and confirmable to users

Dulenchik commented:
@b-artem there are few bugs or strange behavior (for example users, who sign in through social network with email should not confirm it), but anyway good job 👍

b-artem committed:
no more email confirmation when sign in through social network, stylesheet optimization
